mutator-common/src/display_report.rs

```diff
@@ -251,6 +342,73 @@ fn find_mutated_line_number(file_diff: &str) -> Result<usize> {
     Ok(current_line_no)
 }
 
+/// Displays mutants in a readable format.
+pub fn display_mutants_on_screen(
+    path_to_report: impl AsRef<Path>,
+    modules: &ModuleFilter,
+    functions: &FunctionFilter,
+    mutant_filter: &MutantFilter,
+) -> Result<()> {
+    let report = Report::load_from_json_file(path_to_report.as_ref())?;
+    let files_to_print = modules.get_all_files_containing_the_modules(&report);
+    let Report { mut files, .. } = report;
+
+    if files_to_print.is_empty() {
+        println!("No matching files found.");
+        return Ok(());
+    };
+
+    let mut all_mutant_stats = Vec::<MutantStats>::new();
+    files_to_print.into_iter().for_each(|file| {
+        let mut file_mutant_stats = files.remove(&file).expect("can't fetch the mutant stats");
+
+        if let FunctionFilter::Selected(filtered_funcs) = functions {
+            file_mutant_stats
+                .into_iter()
+                .filter(|m| {
+                    let (_module, func) = m
+                        .module_func
+                        .split_once("::")
+                        .expect("invalid function signature in the report file");
+                    filtered_funcs.contains(&func.to_owned())
+                })
+                .for_each(|m| all_mutant_stats.push(m));
+        } else {
+            all_mutant_stats.append(&mut file_mutant_stats);
+        }
+    });
```

Comment:
To avoid nested loops, we could use something like the below code. Additionally, using `retain` modifies `file_mutant_stats` in place when filtering functions, making it more memory efficient.
I've not compiled below - it's just PoC 😄 

```suggestion
    for file in files_to_print {
        if let Some(mut file_mutant_stats) = files.remove(&file) {
            match functions {
                FunctionFilter::Selected(filtered_funcs) => {
                    file_mutant_stats.retain(|m| {
                        let (_, func) = m
                            .module_func
                            .split_once("::")
                            .expect("invalid function signature in the report file");
                        filtered_funcs.contains(&func.to_owned())
                    });
                }
                _ => {}
            }
            all_mutant_stats.extend(file_mutant_stats);
        }
    }
```

Reply:
Done

Reply:
`retain` and `extend` are quite nice. thank you!
and clippy helped with making it even shorter.